Difficulty level

DeepSky8 opened the pull request:
This branch adds support for a third data point when creating a new issue. This data point is titled Difficulty. No explanation is made for what this means.

Radio buttons ensure that the issue creator will pick from appropriate difficulty levels.

Ideally, the issue difficulty can be set by users at a time after issue creation. This has not yet been implemented.

There are two lines of legacy code currently commented out. These lines have not yet been deleted to provide a jumping off point if radio buttons prove unsuitable to the client.

DeepSky8 committed:
Almost functional code to set a difficulty for a new issue.
Throws error when creating actual issue.
Does not yet ensure it is a valid integer value for difficulty.

DeepSky8 committed:
Gotta have that crucial default constructor for an Issue. Issue.html, line 18.

DeepSky8 committed:
A few minor changes that improve functionality, including default constructor.

DeepSky8 committed:
Added radio-button functionality to the newIssue form.

bkyrlach commented on src/main/java/com/kyrlach/issuetracker/issue/Issue.java:
```diff
@@ -13,12 +13,14 @@
     public Issue() {}
     
-    public Issue(String title, String description) {
+    public Issue(String title, String description, String difficulty) {
```
Shouldn't this be an int?

bkyrlach commented on src/main/java/com/kyrlach/issuetracker/issue/IssueForm.java:
```diff
@@ -3,6 +3,7 @@
 public class IssueForm {
 	private String title;
 	private String description;
+	private String difficulty;
```
Shouldn't this be an int?

bkyrlach commented on src/main/resources/templates/newIssue.html:
```diff
@@ -16,6 +16,32 @@
 	    		<label for="description">Description</label>
 	    		<textarea th:field="*{description}"></textarea>
 	    		
+	    		<label for="difficulty">Difficulty</label>
```
What happens if they don't select a difficulty level?

DeepSky8 committed:
Updated IssueForm to include a default difficulty level if
the user does not assign a difficulty level via the radio buttons.
Updated the definition of Issue to (String, String, Number)
